### ISSUE TITLE
Print all pages

### ISSUE THREAD

**Issue by avurro** [ygimenez/Pagination-Utils#8]
**Is your feature request related to a problem? Please describe.**
Both the pagination and the categorization become hateful when the pages become many.

**Describe the solution you'd like**
It would be great to add (optionally) an emoji that allows sequential printing of all pages.

**Comment by ygimenez** [comment]
Unfortunately, the very reason for using pagination is how limited embed space is. You could, depending on the amount of data, put everything in one big embed, but you'll be capped really quick (trust me, really quick). 

You kinda can create what you're asking by using `Pages.buttonize()` to create a replica of the paginate method and add an extra button for showing the big embed, but again, you can put only so much data in it.

**Comment by avurro** [comment]
This is not what I asked for.
what I ask is an emoji that allows you to print all the created embeds in sequence, not to combine them into one.

**Comment by avurro** [comment]
Why do you close the issues without waiting for applicant's response? 😕

**Comment by ygimenez** [comment]
You're always free to reopen it, I closed because I understood you wanted all the pages in a single embed.
Also, this is the first issue I closed without waiting response.

**Comment by ygimenez** [comment]
Discord doesn't allow more than one embed per message, this is an API limit.
Sending multiple messages would fall into API abuse due to ratelimits.

**Comment by avurro** [comment]
![image](https://user-images.githubusercontent.com/2487392/104472957-c60ebf80-55bc-11eb-8f95-6998f3d02cef.png)
As you can see it is possible instead.

![image](https://user-images.githubusercontent.com/2487392/104473135-f2c2d700-55bc-11eb-8580-1ed131c5ab27.png)
Pressing the :scroll: emoji prints all pages in sequence for this bot.

Can you replicate this feature?

**Comment by ygimenez** [comment]
![image](https://user-images.githubusercontent.com/21084778/104592999-d15dfb80-564d-11eb-8c1a-415683bbaf3a.png)
As I said, the only way to do that is by sending multiple messages, with one embed per message (only webhooks can send 10/message), which will easily become api abuse if you have many pages.

**Comment by avurro** [comment]
> As I said, the only way to do that is by sending multiple messages, with one embed per message

But that's what I'm asking from the title of this issue: Print all pages, which does not mean print all in one block, but print all embeds.

> which will easily become api abuse if you have many pages.

This bot print 5 embeds in sequence and there aren't problems.

Do you think you can do the same?

**Comment by ygimenez** [comment]
> But that's what I'm asking from the title of this issue: Print all pages, which does not mean print all in one block, but print all embeds.

Which I answered right after I reopened the issue.
> Do you think you can do the same?

I definitely can do the same, but again, 5 pages is one story, 50 pages is a whole different story (you said "...when the pages become **many**"). 
The current ratelimit, if I recall correctly, is 5 messages per 5 seconds per channel, which means if you had to print 50 pages it'd take roughly 50 seconds, resulting in a massive spam if someone clicks twice or more.

This would effectively DDoS your bot, or even crash Discord servers (nuke).

**Comment by avurro** [comment]
I understand what you are saying, but I assure you that this bot writes all the pages instantly, without any kind of delay.

This implementation would make your library complete.

Release it and let's do some tests, what do you think about?

**Comment by ygimenez** [comment]
It does print instantly because it sacrifices quantity over speed. It's impossible to circumvent Discord ratelimits, you can see it in action yourself by putting a `.sendMessage("test")` inside a loop then observing the huge delay after a few messages.

**Comment by avurro** [comment]
You can add this feature with a waning: "in case of numerous pages, discord rate limits will take over".

**Comment by ygimenez** [comment]
By numerous I mean anything above 5 pages, it makes no sense to implement such limited feature.
You can make this feature yourself by using `Pages.buttonize()` to execute a loop which prints each page, that method gives you the freedom to create any custom button.

TL;DR: This feature is considered API abuse, will not be implemented.

**Comment by avurro** [comment]
It will mean that I clone the project and implement it myself.
 
Dozens of bots implement this feature, having a limitation in print does not mean going to jail if it occurs :expressionless:

**Comment by ygimenez** [comment]
Sure, you're free to implement it yourself, just don't blame me if you get your account banned 😄

**Comment by avurro** [comment]
There is no ban, Discord simply blocks text sharing. Don't be afraid man! :punch:

**Comment by Cabbasca** [comment]
Discord allows up to 10 embeds in one message

**Comment by ygimenez** [comment]
> Discord allows up to 10 embeds in one message

They did add that functionality to bots (previously limited to webhooks) recently, but although it's now possible it's outside the scope of the library (which is fragmenting data into pages and adding a framework for interactive buttons).

That said, this is achievable by simply sending the pages directly, instead of calling `Pages.paginate`.

If there's a higher demand for this feature i might include it in a future release.
